src/server/multi_test.cc

```diff
@@ -715,4 +718,32 @@ TEST_F(MultiTest, ContendedList) {
   EXPECT_EQ(Run({"llen", "chan-2"}), "0");
 }
 
+// Test that squashing makes single-key ops atomic withing a non-atomic tx
+// because it runs them within one hop.
+TEST_F(MultiTest, TestSquashing) {
```

Comment:
we have another problem with blpop: if we have a pending blpop then `lpush`, `lpop` sequence will skip it.

Reply:
One workaround for this could be - to blacklist any list removing commands for squashing.

Reply:
It is also the case that if we have successive LPUSH'es, then the next waking up BLPOP will pop the last value? 🤔

Reply:
yes, for multi/exec it should be correct because we should execute each "exec" atomically.